File: webflux-dialogue/src/testFixtures/java/com/study/webflux/rag/fixture/ConversationTurnFixture.java
package com.study.webflux.rag.fixture;

import java.time.Instant;

import com.study.webflux.rag.domain.dialogue.model.ConversationSessionId;
import com.study.webflux.rag.domain.dialogue.model.ConversationTurn;

public final class ConversationTurnFixture {

	public static final String DEFAULT_QUERY = "안녕하세요";
	public static final String DEFAULT_RESPONSE = "반갑습니다";

	private ConversationTurnFixture() {
	}

	public static ConversationTurn create() {
		return ConversationTurn.create(ConversationSessionFixture.createId(), DEFAULT_QUERY);
	}

	public static ConversationTurn create(ConversationSessionId sessionId) {
		return ConversationTurn.create(sessionId, DEFAULT_QUERY);
	}

	public static ConversationTurn create(String query) {
		return ConversationTurn.create(ConversationSessionFixture.createId(), query);
	}

	public static ConversationTurn createWithResponse() {
		return ConversationTurn.create(ConversationSessionFixture.createId(), DEFAULT_QUERY)
			.withResponse(DEFAULT_RESPONSE);
	}

	public static ConversationTurn createWithResponse(ConversationSessionId sessionId) {
		return ConversationTurn.create(sessionId, DEFAULT_QUERY).withResponse(DEFAULT_RESPONSE);
	}

	public static ConversationTurn createWithId(String id) {
		return ConversationTurn.withId(id,
			ConversationSessionFixture.createId(),
			DEFAULT_QUERY,
			DEFAULT_RESPONSE,
			Instant.now());
	}
}

File: app-miyou/src/testFixtures/kotlin/com/miyou/app/fixture/ConversationTurnFixture.kt
package com.miyou.app.fixture

import com.miyou.app.domain.dialogue.model.ConversationSessionId
import com.miyou.app.domain.dialogue.model.ConversationTurn
import java.time.Instant

object ConversationTurnFixture {

	const val DEFAULT_QUERY = "안녕하세요"
	const val DEFAULT_RESPONSE = "반갑습니다"

	@JvmStatic
	fun create(): ConversationTurn = ConversationTurn.create(ConversationSessionFixture.createId(), DEFAULT_QUERY)

	@JvmStatic
	fun create(sessionId: ConversationSessionId): ConversationTurn =
		ConversationTurn.create(sessionId, DEFAULT_QUERY)

	@JvmStatic
	fun create(query: String): ConversationTurn =
		ConversationTurn.create(ConversationSessionFixture.createId(), query)

	@JvmStatic
	fun createWithResponse(): ConversationTurn =
		ConversationTurn.create(ConversationSessionFixture.createId(), DEFAULT_QUERY)
			.withResponse(DEFAULT_RESPONSE)

	@JvmStatic
	fun createWithResponse(sessionId: ConversationSessionId): ConversationTurn =
		ConversationTurn.create(sessionId, DEFAULT_QUERY).withResponse(DEFAULT_RESPONSE)

	@JvmStatic
	fun createWithId(id: String): ConversationTurn =
		ConversationTurn.withId(
			id,
			ConversationSessionFixture.createId(),
			DEFAULT_QUERY,
			DEFAULT_RESPONSE,
			Instant.now(),
		)
}
